homeassistant/components/conversation/default_agent.py

```diff
@@ -592,6 +610,86 @@ def _get_error_text(
         response_str = lang_intents.error_responses.get(response_key)
         return response_str or _DEFAULT_ERROR_TEXT
 
+    def register_trigger(
+        self,
+        sentences: list[str],
+        callback: TRIGGER_CALLBACK_TYPE,
+        response: str | None = None,
+    ) -> core.CALLBACK_TYPE:
+        """Register a list of sentences that will trigger a callback when recognized."""
+        action = TriggerAction(callback, response=response)
+        for sentence in sentences:
+            self._trigger_sentences[sentence].append(action)
```

Comment:
Why would we support matching the same sentence twice?

Reply:
A user could reuse the same sentence in a separate trigger. It's odd, but having a trigger "use up" a sentence seemed odd too.

Reply:
The problem is that `hello` and `hello[s]` are 2 different keys in `self._trigger_sentences` but both would match the sentence "Hello", yet not both will be called in that case.

Reply:
Using the trigger id as the key now (and the virtual intent name for hassil)